flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -322,11 +323,7 @@ Resource getContainerResource() {
 	public boolean stopWorker(final YarnWorkerNode workerNode) {
 		final Container container = workerNode.getContainer();
 		log.info("Stopping container {}.", container.getId());
-		try {
-			nodeManagerClient.stopContainer(container.getId(), container.getNodeId());
-		} catch (final Exception e) {
-			log.warn("Error while calling YARN Node Manager to stop container", e);
-		}
+		nodeManagerClient.stopContainerAsync(container.getId(), container.getNodeId());
```

Comment:
How can we react to stop container failure?

Reply:
Well I notice `onXXX` methods below. Would you explain why we don't

```java
try {
  ...
} catch (Throwable t) {
  onStopContainerError(...)
}
```

here as we do when start container?

Reply:
`stopContainerAsync` will not throw any exception. If stop failed, `onStartContainerError` will be called.
We catch the exception when 'startContainerAsync' just because `createTaskExecutorLaunchContext` may throw an exception.

Reply:
Thanks for your explanation. From the signature of `startContainerAsync` it is also not marked throws exceptions.

Reply:
The exception is not thrown by `startContainerAsync `, but the `createTaskExecutorLaunchContext `.

Reply:
got it. Sorry for no looking around :-(